consensus/ibft/ibft.go

```diff
@@ -388,6 +395,12 @@ func (i *Ibft) buildBlock(snap *Snapshot, parent *types.Header) (*types.Block, e
 		if txn == nil {
 			break
 		}
+		if gas := transition.AvailableGas(); gas < state.TxGas {
+			i.logger.Info("Block doesn't have enough gas to process new transaction", "remaining", gas, "minimum required gas", state.TxGas)
+			retFn()
+			break
+		}
```

Comment:
What do you think of moving this check and the one in `dev.go` into `Write()` method of `executor.go`'s Transition object?
That way we will end up with only one check instead of two.

Reply:
I think this is great idea since can do this check when other node appends new block.